doc/reference/networking/openthread_integration.rst

```diff
@@ -0,0 +1,172 @@
+The drawback, on the other hand, lies in the reception path in which the IP packet
+needs to traverse the OpenThread L3 layer and then Zephyr's L3 layer in order to 
+reach the BSD socket.
+
+Note that if one desires the application can use just the OpenThread API 
+with it's IPv6 stack and even the internal CoAP implementation 
+which is the application protocol.
+
+File system and shim layer
+**************************
+
+The OpenThread Network Stack is present in the following paths:
+- OpenThread stack location: modules/lib/openthread/
+- OpenThread shim layer location: zephyr/subsys/net/lib/openthread/platform/
+It's responsibilities include:
+- translating the data into the Zephyr's native net_pkt,
+- providing openthread thread body and synchronization API,
+- providinng openthread_send(), openthread_recv() calls which are registered as the L2 interface API,
+- providing a way to initialize the OpenThread stack,
+- implementing callback functions used by the OpenThread stack
+
+The nRF IEEE802.15.4 Radio Driver is present in the following paths:
+- nRF IEEE802.15.4 Radio Driver shim layer location: zephyr/drivers/ieee802154/{ieee802154_nrf5.c/ieee802154_nrf5.h
+- nRF IEEE802.15.4 Radio Driver location: modules/hal/nordic/drivers/nrf_radio_802154
+The RX connection of the radio driver is done with the interrupt handler 
+registered using the Zephyr's mechanism with the NRF_802154_INTERNAL_RADIO_IRQ_HANDLING=0 defined. 
+The registered IRQ handler uses Zephyr's FIFO to pass the IEEE802.15.4 frame further.
+The 802154 thread runs on the highest cooperative priority and waits on this FIFO. 
+Once new frame apears it continues with the processing.
+
+The TX connection uses the workqueue which calls the radio driver calls to schedule the transmission.
+Then the RTC IRQ is used to send the frame over the air.
+
+Threads
+*******
+- openthread - responsible for receiving IEEE 802.15.4 frames during reception.
+  When the reassembled trafic turns out to be the IPv6 packet it calls the ot_receive_handler()
+  which injects it back to the L2 via the net_recv_data() so it could later reach the Zephyr's IP stack.
+  During the transmission it's job is to handle the previously scheduled OpenThread Tasklet containig 
+  a message to be sent.
+
+- rx_workq - responsible for receiving L2 frames and directing them either to the openthread process
+  or the Zephyr's IP stack during reception depening if it is the IEEE 802.15.4 frame
+  or the IPv6 packet.
+
+- tx_workq - responsible for receiving UDP packet and scheduling the OpenThread Tasklet
+  for transmission and unlocking the openthread thread by giving the semaphore.
+
+- workqueue - responsible for invoking the radio driver API in order to schedule a transmission.
+
+- 802154 RX - resposible for the "upper half" processing of the radio frame reception. 
+  Works on the objects of type nrf5_802154_rx_frame which are put to the nrf5_data.rx_fifo
```

Comment:
```suggestion
  Works on the objects of type ``nrf5_802154_rx_frame`` which are put to the ``nrf5_data.rx_fifo``
```
Please make sure all code inline content is highlighted this way. More info: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/documentation/index.html#content-highlighting